datasets/dataset_cifar10.py

```diff
@@ -27,13 +33,35 @@ def preprocess_dataset_inputs(x):
 
 # Data samples pre-processing method for labels
 def preprocess_dataset_labels(y):
-    y = keras.utils.to_categorical(y, num_classes)
+    y = to_categorical(y, num_classes)
 
     return y
 
 
 def generate_new_dataset():
-    (x_train, y_train), (x_test, y_test) = cifar10.load_data()
+    attempts = 0
+    while True:
+        try:
+            (x_train, y_train), (x_test, y_test) = cifar10.load_data()
+            break
+        except HTTPError as e:
+            logger.debug(
+                f'URL fetch failure on https://www.cs.toronto.edu/~kriz/cifar-10-python.tar.gz : {e.code} -- {e.msg}')
+            if attempts < constants.NUMBER_OF_DOWNLOAD_ATTEMPTS:
+                sleep(2)
+                attempts += 1
+            else:
+                raise
+        except URLError as e:
+            logger.debug(
+                f'URL fetch failure on https://www.cs.toronto.edu/~kriz/cifar-10-python.tar.gz : '
+                f'{e.errno} -- {e.reason}')
+            if attempts < constants.NUMBER_OF_DOWNLOAD_ATTEMPTS:
+                sleep(2)
+                attempts += 1
+            else:
+                raise
+
```

Comment:
Nice 👍

Reply:
Question : as the only thing that differs in handling the `HTTPError` and the `URLError` is the debug log message, can't it be factorized?

Reply:
HTTP et URLError don't have the same attributs (even if .msg and .reason are basically the samefor HTTP, and HTTP have a .errno, but without interesting infos (None value)
I factorized that

Reply:
Ok, clear and clean